pkg/cmd/flags/logger.go

```diff
@@ -64,40 +130,96 @@ func PrepareLogger(logOptions []string, newBinary bool) (logger.LoggingConfig, e
 			if !strings.HasSuffix(opt, "aggregate") {
 				vals := strings.Split(opt, ":")
 				if len(vals) != 2 || len(vals[1]) <= 1 {
-					return logger.LoggingConfig{}, InvalidLogOption(opt, newBinary)
+					return logger.LoggingConfig{}, InvalidLogOptionValue(opt, newBinary)
 				}
 
 				// handle only seconds and minutes
 				timeSuffix := vals[1][len(vals[1])-1:][0]
 				if timeSuffix != 's' && timeSuffix != 'm' {
-					return logger.LoggingConfig{}, InvalidLogOption(opt, newBinary)
+					return logger.LoggingConfig{}, InvalidLogOptionValue(opt, newBinary)
 				}
 				prevByte := vals[1][len(vals[1])-2:][0]
 				if timeSuffix == 's' && !unicode.IsDigit(rune(prevByte)) {
-					return logger.LoggingConfig{}, InvalidLogOption(opt, newBinary)
+					return logger.LoggingConfig{}, InvalidLogOptionValue(opt, newBinary)
 				}
 
 				interval, err = time.ParseDuration(vals[1])
 				if err != nil {
-					return logger.LoggingConfig{}, InvalidLogOption(opt, newBinary)
+					return logger.LoggingConfig{}, InvalidLogOptionValue(opt, newBinary)
 				}
 			}
+
 			agg = true
 			continue
 		}
 
-		switch opt {
-		case "debug":
-			lvl = logger.DebugLevel
-		case "info":
-			lvl = logger.InfoLevel
-		case "warn":
-			lvl = logger.WarnLevel
-		case "error":
-			lvl = logger.ErrorLevel
-		case "fatal":
-			lvl = logger.FatalLevel
-		default:
+		// parse filter option
```

Comment:
can we extract this section to be its own method?

Reply:
I'll take a look at this later. 👍🏼

Reply:
Well, we can, but it would demand some back and forth of data and state. So let's leave it rest for now.